webviz_subsurface/plugins/_co2_leakage/_utilities/initialization.py

```diff
@@ -60,23 +62,84 @@ def init_surface_providers(
 
 
 def init_well_pick_provider(
-    well_pick_dict: Dict[str, Optional[str]],
+    ensemble_paths: Dict[str, str],
+    well_pick_path: Optional[str],
     map_surface_names_to_well_pick_names: Optional[Dict[str, str]],
-) -> Dict[str, Optional[WellPickProvider]]:
-    well_pick_provider: Dict[str, Optional[WellPickProvider]] = {}
-    ensembles = list(well_pick_dict.keys())
-    for ens in ensembles:
-        well_pick_path = well_pick_dict[ens]
-        if well_pick_path is None:
-            well_pick_provider[ens] = None
-        else:
-            try:
-                well_pick_provider[ens] = WellPickProvider(
-                    read_csv(well_pick_path), map_surface_names_to_well_pick_names
-                )
-            except OSError:
-                well_pick_provider[ens] = None
-    return well_pick_provider
+) -> Dict[str, EnsembleWellPicks]:
+    if well_pick_path is None:
+        return {}
+
+    return {
+        ens: EnsembleWellPicks(
+            ens_p, well_pick_path, map_surface_names_to_well_pick_names
+        )
+        for ens, ens_p in ensemble_paths.items()
+    }
+
+
+def init_hazardous_boundary_providers(
+    ensemble_paths: Dict[str, str],
+    poly_path: Optional[str],
+) -> Dict[str, Optional[EnsemblePolygonProvider]]:
+    if poly_path is None:
+        return {}
+
+    return {
+        ens: _init_hazardous_boundary_provider(ens_path, poly_path)
+        for ens, ens_path in ensemble_paths.items()
+    }
+
+
+def _init_hazardous_boundary_provider(
+    ensemble_path: str, poly_path: str
+) -> Optional[EnsemblePolygonProvider]:
+    try:
+        return EnsemblePolygonProvider(
+            ensemble_path,
+            poly_path,
+            "Hazardous Polygon",
+            "hazardous-boundary-layer",
+            [200, 0, 0, 120],
+        )
+    except OSError as e:
+        LOGGER.warning(
+            f"Failed to create hazardous boundary provider for ensemble path:"
+            f" '{ensemble_path}' and poly path '{poly_path}': {e}"
+        )
+        return None
+
+
+def init_containment_boundary_providers(
+    ensemble_paths: Dict[str, str],
+    poly_path: Optional[str],
+) -> Dict[str, Optional[EnsemblePolygonProvider]]:
+    if poly_path is None:
+        return {}
+
+    return {
+        ens: _init_containment_boundary_provider(ens_path, poly_path)
+        for ens, ens_path in ensemble_paths.items()
+    }
+
+
+def _init_containment_boundary_provider(
+    ensemble_path: str,
+    poly_path: str,
+) -> Optional[EnsemblePolygonProvider]:
+    try:
+        return EnsemblePolygonProvider(
+            ensemble_path,
+            poly_path,
+            "Containment Polygon",
+            "license-boundary-layer",
+            [0, 172, 0, 120],
+        )
+    except OSError as e:
+        LOGGER.warning(
+            "Failed to create containment boundary provider for ensemble path:"
+            f" '{ensemble_path}' and poly path '{poly_path}': {e}"
+        )
+        return None
```

Comment:
Could we combine these 4 last methods above to 2 methods? And have extra input "type" that is either containment or haz... Currently init_containment_boundary_providers and init_hazardous_boundary_providers are almost identical.
Not a big issue, the methods are quite small, so whatever you feel is best is fine